exporter/exporterhelper/internal/producer_consumer_queue.go

```diff
@@ -12,21 +14,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package exporterhelper
+package internal
 
-// consumersQueue is largely based on queue.BoundedQueue and matches the subset used in the collector
-// It describes a producer-consumer exchange which can be backed by e.g. the memory-based ring buffer queue
-// (queue.BoundedQueue) or via disk-based queue (persistentQueue)
-type consumersQueue interface {
+// Consumer consumes data from a bounded queue
+type Consumer interface {
```

Comment:
Is this used anywhere?

Reply:
It is used here https://github.com/open-telemetry/opentelemetry-collector/pull/4115/files#diff-e3cf80389040c560d0447ca06b063cbe062967fe24772e2b1d40cd5c696226b7R59-R61 I added that interface to the Jaeger codebase a while ago :)

Reply:
After we move all to "internal" (PR part 2) - we could unexport it, I think

Reply:
Yes, sorry unused outside the package. No other public API exposes it.

Reply:
Currently it's being used here: https://github.com/pmm-sumo/opentelemetry-collector/blob/pq-move-to-internal-part1/exporter/exporterhelper/persistent_queue.go#L51-L53 - so if that works we could keep it as it is here and I will update the second PR changing `Consumer` to `consumer`